egcg_core/util.py

```diff
@@ -26,7 +26,10 @@ def find_files(*path_parts):
 
 def find_file(*path_parts):
     files = find_files(*path_parts)
-    if files:
+    nfiles = len(files)
+    if nfiles > 1:
+        app_logger.warning('Searched pattern %s for one file, but got %s', path_parts, nfiles)
+    elif nfiles == 1:
```

Comment:
This changes the behaviour.
We should check that it is not used with the implicit behaviour that it returns the first element when many are found

Reply:
I've done a search through all usages of `find_file` within this organisation, and for all of them I'd expect to find only one file and would be concerned if there were multiple matches. Not the way `find_file` was originally written, but it's how we've been using it